spec/master_spec.js

```diff
@@ -0,0 +1,160 @@
+var noop = require("lodash.noop"),
+    cluster = require("cluster"),
+    runMasterJasmine = require("../lib/master");
+
+describe("master", function() {
+  beforeEach(function() {
+    var fakeRunJasmine = this.fakeRunJasmine = jasmine.createSpy("fakeRunJasmine");
+    spyOn(require("module"), "_load").and.callFake(function() {
+      if (arguments[0] === "./run") {
+        return fakeRunJasmine;
+      } else {
+        return this._load.and.originalFn.apply(this, arguments);
+      }
+    });
+
+    var fakeSend = this.fakeSend = jasmine.createSpy("fakeSend");
+    spyOn(cluster, "fork").and.returnValue({
+      send: fakeSend
+    })
+
+    // this.fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"])
+    // this.fakeJasmine.specFiles = [
+    //   "./master_spec.js",
+    //   "./worker_spec.js"
+    // ];
+  });
+
+  it("massege callback 'jasmineDone' should send and kill properly", function() {
+    var fakeJasmine = jasmine.createSpyObj('jasmine', ["execute", "configureDefaultReporter", "addSpecFiles"])
+
+    fakeJasmine.specFiles = [
+      "./master_spec.js",
+      "./worker_spec.js"
+    ];
+    fakeJasmine.reporter = jasmine.createSpyObj('reporter', ['jasmineDone']);
+    var fakeSend = jasmine.createSpy("fakeSend"),
+        fakeKill = jasmine.createSpy("fakeKill"),
+        env = { 
+          workerCount: 1,
+          files: []
+        },
+        allArgs = null;
+    cluster.removeAllListeners("message");
+    runMasterJasmine(fakeJasmine, env, console.log);
+
+    cluster.emit("message", { send: fakeSend, kill: fakeKill }, { kind: "jasmineDone" })
+    expect(fakeJasmine.reporter.jasmineDone).not.toHaveBeenCalled();
+    expect(fakeKill).not.toHaveBeenCalled();
+    allArgs = (fakeSend.calls.allArgs());
+    expect(allArgs[0][0].workerCount).toBe(env.workerCount);
+    expect(allArgs[0][0].files[0]).toBe(fakeJasmine.specFiles[1]);
+    expect(allArgs[0][0].seed).toBe(env.seed);
+
+    cluster.emit("message", { send: fakeSend, kill: fakeKill }, { kind: "jasmineDone" })
+    expect(fakeKill).toHaveBeenCalled();
+    expect(fakeSend).toHaveBeenCalledTimes(1);
+  })
+
+  describe("reporter callback", function() {
+    beforeEach(function() {
+      cluster.removeAllListeners("message");
+      cluster.removeAllListeners("exit");
+    });
+
+    it("'specStarted', 'specDone', 'suiteStarted', 'suiteDone' should forward the payload directly", function() {
```

Comment:
I would like to see this broken out into individual specs for each callback or have the spies be reset between executions and the continued `toHaveBeenCalledWith` can't break currently because the previous call is still there.